diff --git a/frmts/pdf/pdfsdk_headers.h b/frmts/pdf/pdfsdk_headers.h
index d5a3af3dea..e587a25acd 100644
--- a/frmts/pdf/pdfsdk_headers.h
+++ b/frmts/pdf/pdfsdk_headers.h
@@ -37,6 +37,11 @@
 
 #ifdef HAVE_POPPLER
 
+// The "#define private public" hacks we do below do not play well with the
+// sstream header.
+// Cf https://github.com/conda-forge/gdal-feedstock/pull/1019#issuecomment-2528710051
+#include <sstream>
+
 /* Horrible hack because there's a conflict between struct FlateDecode of */
 /* include/poppler/Stream.h and the FlateDecode() function of */
 /* pdfium/core/include/fpdfapi/fpdf_parser.h. */
